data/questions_gen_postgres.csv

```diff
@@ -44,14 +44,14 @@ How many courses are offered for each semester id?,"SELECT course_offering.semes
 "What is the total number of students enrolled in each course, ordered from highest to lowest?","SELECT {course.course_id, course.name, course.number}, SUM(course.num_enrolled) AS total_students FROM course GROUP BY {} ORDER BY total_students DESC NULLS LAST;",advising,order_by,
 "What is the total number of credits earned by each student, ordered from highest to lowest? Give the student id and the total number of credits.","SELECT student.student_id, student.total_credit FROM student ORDER BY student.total_credit DESC NULLS LAST;",advising,order_by,
 "What is the name of the instructor who has taught the most courses, and how many courses have they taught?","SELECT instructor.name, count(offering_instructor.offering_id) AS num_courses FROM offering_instructor JOIN instructor ON offering_instructor.instructor_id = instructor.instructor_id GROUP BY instructor.name ORDER BY num_courses DESC LIMIT 1;",advising,order_by,
-What is the ratio of the total number of students enrolled in courses with exams to the total number of students enrolled in courses without exams?,"WITH exams AS (SELECT DISTINCT sr.student_id FROM public.student_record sr JOIN public.course_offering co ON sr.offering_id = co.offering_id WHERE co.has_final_exam = TRUE ), no_exams AS (SELECT DISTINCT sr.student_id FROM public.student_record sr JOIN public.course_offering co ON sr.offering_id = co.offering_id WHERE co.has_final_exam = FALSE ) SELECT (SELECT COUNT(student_id) FROM exams)::FLOAT / (SELECT COUNT(student_id) FROM no_exams) AS ratio;",advising,ratio,
+What is the ratio of the total number of students enrolled in courses with exams to the total number of students enrolled in courses without exams?,"SELECT SUM(CASE WHEN c.has_exams THEN c.num_enrolled ELSE 0 END)::FLOAT / SUM(CASE WHEN NOT c.has_exams THEN c.num_enrolled ELSE 0 END) AS ratio FROM course c;",advising,ratio,

```

Comment:
oo thanks for pointing out the slightly ambiguous reference of final_exam vs exam!